src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1032,7 +1032,7 @@ public bool TakeStartLine(ReadableBuffer buffer, out ReadCursor consumed, out Re
             var result = _parser.ParseRequestLine(this, buffer, out consumed, out examined);
             if (!result && overLength)
             {
-                RejectRequest(RequestRejectionReason.RequestLineTooLong);
+                RequestRejectionUtilities.RejectRequest(RequestRejectionReason.RequestLineTooLong);
```

Comment:
Note: in the pre-complied framework portion it does move all the cold exception code to a different part of the assembly so all the hot code is kept together. However at runtime jit it can't alter the layout of the executable code to achieve this; as it generates the code in-place without knowing the holistic picture to alter the assembly layout. So its better to manually separate out the cold code keeping the hot code together in the instruction cache.